gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcHeaderInterceptor.java

```diff
@@ -36,19 +36,18 @@
 import io.grpc.ForwardingClientCall.SimpleForwardingClientCall;
 import io.grpc.Metadata;
 import io.grpc.MethodDescriptor;
+import java.util.Map;
 
 /**
- * An intercepter to handle custom header.
+ * An intercepter to handle custom headers.
  *
  * <p>Package-private for internal usage.
  */
 class GrpcHeaderInterceptor implements ClientInterceptor {
-  private static final Metadata.Key<String> HEADER_KEY =
-      Metadata.Key.of("x-goog-api-client", Metadata.ASCII_STRING_MARSHALLER);
-  private final String header;
+  private final Map<String, String> headers;
```

Comment:
Please rename this so it does not conflict with `start(ClientCall.Listener<RespT> responseListener, Metadata headers)` parameter `headers` name (and you would not need explicit `GrpcHeaderInterceptor.this.` prefix).